django/db/models/base.py

```diff
@@ -1108,6 +1108,26 @@ def _prepare_related_fields_for_save(self, operation_name, fields=None):
                 ):
                     field.delete_cached_value(self)
 
+        # GenericForeignKey fields are private
+        for field in self._meta.private_fields:
+            if (
+                field.is_relation
+                and field.is_cached(self)
+                and hasattr(field, "fk_field")
+            ):
+                obj = field.get_cached_value(self, default=None)
+                if not obj:
+                    continue
+                if obj.pk is None:
+                    raise ValueError(
```

Comment:
Should we remove the object from a cache? :thinking:

Reply:
https://github.com/django/django/blob/6f311c7e35090957dd93783b71c5d3af7d5329d4/tests/one_to_one/tests.py#L153
In `OneToOneTest.test_unsaved_object` there's a check around it being removed from the cache
I'm not sure if it is needed in this case? I wouldn't know how to structure the test for the behaviour we want.
But it may be needed, I'm not sure.

Reply:
As far as I'm aware it's unneeded, because a reverse relation `GenericRelation` returns multiple objects, so a single `tag` is not assigned to an `object`.